### PR TITLE
[CT] Fix weight processing for FP8 Marlin

### PR DESCRIPTION
**Pull request by dsikka**
## Purpose
- Current integration pathway is causing gibberish as a result of this commit: 55d037e2e5
- Fix post weight loading processing to properly handle params for FP8A16 marlin

## Test Plan
- Basic local smoke testing
```python
    from vllm import LLM, SamplingParams
    import torch

    prompts = [
        "The Swiss Alps are",
        "Brad Marchand is",
        "The Toronto Maple Leafs are"
    ]

    # Create a sampling params object for greedy sampling
    sampling_params = SamplingParams(temperature=0.80, top_p=0.95, max_tokens=40, min_tokens=10)
    model = "nm-testing/fp8_weight_only_tensor-e2e"

    llm = LLM(
        model=model,
        tensor_parallel_size=1
    )
    output = llm.generate(prompts, sampling_params)
    for out in output:
        print(out.outputs[0].text)
```

Before:
```bash
ISTerteschriftteck auteurswitzBoard fixesISTRLLENGTHanstaltERTzelhalbwitzédéwitzeszwitzконоwitzwitzwitzperpwitzegrwitzħvaäswitzatumwitziš portswitzwitz�witz
ídoISTţäsäs włISTäsISTlungwitzěníegründEGwitzrijäsovedesadernualeț lassenäsbbeualeäsarteaneISTwitz firäs mouse janaltungIIfahrenmandeEC
dagDraw})$.eenwitzlungwitzконоktopwitzanstaltäswitzницeg fixedortenLENGTH Stawitzlogging którewitzħкономSIZEanstalt Schrift方 segmentsjquerycheréléesz� #### absent fot电тку
```

After:
```
 a UNESCO World Heritage Site. The stunning mountain range is made up of glaciers, snow-capped peaks, and meandering rivers, making it an incredible destination
 putting his stamp on the Bruins with his play. Marchand has 10 points in his last 11 games, and he is poised to have a breakout year. The Bru
 poised to offer Auston Matthews a five-year contract extension after their 5-2 win over the Ottawa Senators, according to a report from the Globe and Mail.
```

### CODE REVIEW

**Review by yewentao256 — COMMENTED** [review]
Thanks for the work! Could we test with `lm_eval` to show the accuracy?
